std/container/fixed_array.d

```diff
@@ -0,0 +1,504 @@
+    {
+        version(assert) if(j < length) throw new RangeError;
+        store[i .. j] = value;
+    }
+
+    /// ditto
+    void opSliceUnary(string op)()
+        if(op == "++" || op == "--")
+    {
+        mixin(op~"store[];");
+    }
+
+    /// ditto
+    void opSliceUnary(string op)(size_t i, size_t j)
+        if(op == "++" || op == "--")
+    {
+        mixin(op~"store[i .. j];");
+    }
+
+    /// ditto
+    void opSliceOpAssign(string op)(Elementvalue)
+    {
+        mixin("store[] "~op~"= value;");
+    }
+
+    /// ditto
+    void opSliceOpAssign(string op)(Elementvalue, size_t i, size_t j)
+    {
+        mixin("store[i .. j] "~op~"= value;");
+    }
+
+/**
+Forwards to $(LREF insertBack).
+     */
+    void opOpAssign(string op, Stuff)(Stuff stuff)
+        if (op == "~")
+    {
+        insertBack(stuff);
+    }
+
+/**
+Removes all contents from the container. 
+
+Postcondition: $(D empty)
+
+Complexity: $(BIGOH n)
+     */
+    void clear()
+    {
+        length = 0;
+    }
+
+/**
+Releases the underlying store. This FixedArray
+instance is unuseable afterwarts.
+
+     */
+    Store release()
+    {
+        auto result = move(store);
+        return result[0 .. length];
+    }
+
+/**
+Picks one value in an unspecified position in the container, removes
+it from the container, and returns it. FixedArray forwards removeAny to
+removeBack.
+
+Precondition: $(D !empty)
+
+Returns: The element removed.
+
+Complexity: $(BIGOH log(n)).
+     */
+    Element removeAny()
+    {
+        auto result = move(back);
+        removeBack();
+        return result;
+    }
+    /// ditto
+    alias stableRemoveAny = removeAny;
+/**
+Inserts $(D value) to the front or back of the container. $(D stuff)
+can be a value convertible to $(D Element) or a $(XREF2 range, isInputRange, range) of objects convertible
+to $(D Element). The stable version behaves the same, but guarantees that
+ranges iterating over the container are never invalidated.
+
+Params:
+    stuff = Either a $(XREF2 range, isInputRange, range) of elements to insert or an value convertible to $(D Element).
+
+Returns: The number of elements inserted
+
+Complexity: $(BIGOH m * log(n)), where $(D m) is the number of
+elements in $(D stuff)
+     */
+    size_t insertBack(Stuff)(Stuff stuff)
+    if (isImplicitlyConvertible!(Stuff, Element))
+    {
+        if(length == capacity)
+            throw new RangeError("not enough room to insert stuff");
+        store[length] = stuff;
+        length = length + 1;
+        return 1;
+    }
+
+    /// ditto
+    size_t insertBack(Stuff)(Stuff stuff)
+    if(isInputRange!Stuff && isImplicitlyConvertible!(ElementType!Stuff, Element))
+    {
+        static if(hasLength!Stuff)
+        {
+            if(stuff.length > (capacity - length))
+                throw new RangeError("not enough room to insert stuff");
+        }
+            
+        size_t count;
+        foreach(s; stuff)
```

Comment:
use the explicit form, sadly there are possible slowdowns in generic code with foreach